src/app/ReadClient.cpp

```diff
@@ -427,6 +429,31 @@ CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Build
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ReadClient::WakeUp()
+{
+    VerifyOrReturnError(IsIdle(), CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mPeerActivePeriod != System::Clock::kZero, CHIP_ERROR_INCORRECT_STATE);
+    MoveToState(ClientState::Idle);
+    Close(CHIP_ERROR_TIMEOUT);
```

Comment:
Follow-up: MoveToState(ClientState::Idle) is not necessary, Close may use different Error to inform we are waking up the subscription for ICD.